crates/rust-analyzer/src/handlers.rs

```diff
@@ -1033,7 +1033,8 @@ fn add_quick_fixes(
 
     for fix in diagnostics
         .into_iter()
-        .filter_map(|d| d.fix)
+        .filter_map(|d| d.fixes)
+        .flatten()
```

Comment:
```suggestion
        .flat_map(|d| d.fixes)
```

Reply:
I thought about this but `d.fixes` here is an `Option<Vec<Fix>>` so flat_map doesn't seem to work.
One possible way is 
```rust
        .flat_map(|d| match d.fixes {
            Some(fixes) => fixes,
            _ => vec![],
        })
```
and the other is make `fixes` of `Diagnostic` a `Vec<Fix>` which maybe better but requires more changes to existing code to convert None to empty vec.

Reply:
Changed to the the first one for now https://github.com/rust-analyzer/rust-analyzer/pull/8345/commits/6a8018cb01a577673f9cbc4ab48d91d892e92017

Reply:
Ah, I see, that seems good too